src/main/java/org/waarp/common/database/DbAdmin.java

```diff
@@ -76,262 +63,262 @@
     }
 
     /**
      * @return True if the connection is ReadOnly
      */
     public boolean isReadOnly() {
-        return readOnly;
+        return isReadOnly;
     }
 
     /**
-     * @return true
-     *
-     * @deprecated You should not use this function ask for a connection and catch errors instead.
+     * @return the isActive
      */
-    @Deprecated
     public boolean isActive() {
         return true;
     }
 
     /**
      * @param isActive the isActive to set
-     *
-     * @deprecated This will be removed in a future version of WaarpCommon. 
-     * DBCP instegration set active to true.
      */
-    @Deprecated
     public void setActive(boolean isActive) {
-        // Do Nothing
+        if (!isActive) {
+            logger.debug(Thread.currentThread().getStackTrace().toString());
+        }
+        this.isActive = isActive;
     }
 
     /**
      * Validate connection
-     * 
-     * @throws WaarpDatabaseNoConnectionException if a database access errors occurs
-     * 
-     * @deprecated Use {@link #validateConnection()} instead.
+     *
+     * @throws WaarpDatabaseNoConnectionException
      */
-    @Deprecated
     public void validConnection() throws WaarpDatabaseNoConnectionException {
-        validateConnection();
-    }
-
-    /**
-     * Validate connection
-     * 
-     * @throws WaarpDatabaseNoConnectionException if a database access errors occurs
-     */
-    public void validateConnection() throws WaarpDatabaseNoConnectionException {
         try {
-            ds.getConnection().close();
-        } catch (SQLException e) {
-            throw new WaarpDatabaseNoConnectionException("Cannot access database", e);
+            dbModel.validConnection(getSession());
+        } catch (WaarpDatabaseNoConnectionException e) {
+            getSession().setDisActive(true);
+            setActive(false);
+            throw e;
         }
+        getSession().setDisActive(false);
+        setActive(true);
     }
 
     /**
      * Use a default server for basic connection. Later on, specific connection to database for the
      * scheme that provides access to the table R66DbIndex for one specific Legacy could be done.
-     * 
+     *
      * A this time, only one driver is possible! If a new driver is needed, then we need to create a
      * new DbSession object. Be aware that DbSession.initialize should be call only once for each
      * driver, whatever the number of DbSession objects that could be created (=> need a hashtable
      * for specific driver when created). Also, don't know if two drivers at the same time (two
      * different DbSession) is allowed by JDBC.
-     * 
+     *
      * @param model
      * @param server
      * @param user
-     * @param password
+     * @param passwd
      * @throws WaarpDatabaseNoConnectionException
      */
-    public DbAdmin(DbModel model, String server, String user, String password)
+    public DbAdmin(DbModel model, String server, String user, String passwd)
```

Comment:
Why does this method has no retry for database connections like the constructor below ?
I would expect that all constructors behave the same...